src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/AssemblyDependenciesSubTreeProvider.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Imaging.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies
+{
+    /// <summary>
+    /// Provides NuGet packages sub node to global Dependencies project tree node.
+    /// </summary>
+    [Export(typeof(IProjectDependenciesSubTreeProvider))]
+    [AppliesTo(ProjectCapability.DependenciesTree)]
+    internal class AssemblyDependenciesSubTreeProvider : DependenciesSubTreeProviderBase
+    {
+        public const string ProviderTypeString = "AssemblyDependency";
+
+        public readonly ProjectTreeFlags AssemblySubTreeRootNodeFlags
+                            = ProjectTreeFlags.Create("AssemblySubTreeRootNode");
+
+        public readonly ProjectTreeFlags AssemblySubTreeNodeFlags
+                            = ProjectTreeFlags.Create("AssemblySubTreeNode");
+
+        public AssemblyDependenciesSubTreeProvider()
+        {
+            // sucscribe to design time build to get corresponding items
+            UnresolvedReferenceRuleNames = Empty.OrdinalIgnoreCaseStringSet
+                .Add(AssemblyReference.SchemaName);
+            ResolvedReferenceRuleNames = Empty.OrdinalIgnoreCaseStringSet
+                .Add(ResolvedAssemblyReference.SchemaName);
+        }
+
+        public override string ProviderType
+        {
+            get
+            {
+                return ProviderTypeString;
+            }
+        }
+
+        /// <summary>
+        /// Specifies if dependency sub node thinks that it is in error state. Different sub nodes
+        /// can have different conditions for error state.
+        /// </summary>
+        public override bool IsInErrorState
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        private List<ImageMoniker> _nodeIcons = new List<ImageMoniker>
+        {
+            KnownMonikers.Component
+        };
+
+        public override IEnumerable<ImageMoniker> Icons
+        {
+            get
+            {
+                return _nodeIcons;
+            }
+        }
+
+        protected override IDependencyNode CreateRootNode()
```

Comment:
DependenciesSubTreeProviderBase should be abstract with abstract methods, however code below should stay where it is - each provider implementation creates it's own subtree root nodes with different parameters

---
In reply to: [74507861](https://github.com/dotnet/roslyn-project-system/pull/349#discussion_r74507861) [](ancestors = 74507861)